### PR TITLE
add install script

### PR DESCRIPTION
**Pull request by jairad26**

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+      echo_fexists "$HOME/$profile" && break
+    done
+    ;;
+  esac
+}
+
+build_path_str() {
+  local profile="$1" install_dir="$2"
+  if [[ $profile == *.fish ]]; then
+    echo -e "set -gx HYP_CLI \"$install_dir\"\nstring match -r \".hyp\" \"\$PATH\" > /dev/null; or set -gx PATH \"\$HYP_CLI/bin\" \$PATH"
```
Comment (mattjohnsonpint) [inline install.sh:121]:
```suggestion
    echo -e "set -gx HYP_CLI \"$install_dir\"\nstring match -r \".hypermode\" \"\$PATH\" > /dev/null; or set -gx PATH \"\$HYP_CLI/bin\" \$PATH"
```

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+  else
+    install_dir="${install_dir/#$HOME/\$HOME}"
+    echo -e "\n# Hyp CLI\nexport HYP_CLI=\"$install_dir\"\nexport PATH=\"\$HYP_CLI/bin:\$PATH\""
```
Comment (mattjohnsonpint) [inline install.sh:124]:
```suggestion
    echo -e "\n# ${CLI_NAME}\nexport HYP_CLI=\"$install_dir\"\nexport PATH=\"\$HYP_CLI/bin:\$PATH\""
```

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    printf "Please check your profile scripts and environment.\n" >&2
+    exit 1
+  fi
+  return 0
+}
+
+update_profile() {
+  local install_dir="$1"
+  if [[ ":$PATH:" == *":$install_dir:"* ]]; then
+    printf "[3/4] ${DIM}The Hyp CLI is already in \$PATH${RESET}\n" >&2
```
Comment (mattjohnsonpint) [inline install.sh:140]:
```suggestion
    printf "[3/4] ${DIM}The ${CLI_NAME} is already in \$PATH${RESET}\n" >&2
```

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+# Copyright 2024 Hypermode Inc.
+# Licensed under the terms of the Apache License, Version 2.0
+# See the LICENSE file that accompanied this code for further details.
+#
+# SPDX-FileCopyrightText: 2024 Hypermode Inc. <hello@hypermode.com>
+# SPDX-License-Identifier: Apache-2.0
+
+# Run with: curl install.hypermode.com/hyp.sh -sSfL | bash
+
+# Config
```
Comment (mattjohnsonpint) [inline install.sh:12]:
```suggestion
# Config
CLI_NAME="Hypermode CLI"
```

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+  if [ -z "$profile" ]; then
+    printf "No user shell profile found.\n" >&2
+    return 1
+  fi
+
+  export SHOW_RESET_PROFILE=1
+  export PROFILE="~/$(basename "$profile")"
+
+  if grep -q 'HYP_CLI' "$profile"; then
+    printf "[3/4] ${DIM}The Hyp CLI has already been configured in ${PROFILE}${RESET}\n" >&2
```
Comment (mattjohnsonpint) [inline install.sh:154]:
```suggestion
    printf "[3/4] ${DIM}The ${CLI_NAME} has already been configured in ${PROFILE}${RESET}\n" >&2
```

File: install.sh [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+    return 0
+  fi
+
+  local path_str="$(build_path_str "$profile" "$install_dir")"
+  echo "$path_str" >>"$profile"
+
+  printf "[3/4] ${DIM}Added the Hyp CLI to the PATH in ${PROFILE}${RESET}\n" >&2
```
Comment (mattjohnsonpint) [inline install.sh:161]:
```suggestion
  printf "[3/4] ${DIM}Added the ${CLI_NAME} to the PATH in ${PROFILE}${RESET}\n" >&2
```

File: install.sh [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+  case "$VERSION" in
+  latest)
+    VERSION="$(get_latest_version)"
+    ;;
+  *) ;;
+  esac
+
+  install_release
+  if [ "$?" == 0 ]; then
+    update_profile "$INSTALL_DIR" && printf "[4/4] ${DIM}Installed Hyp CLI${RESET}\n" >&2
```
Comment (mattjohnsonpint) [inline install.sh:178]:
```suggestion
    update_profile "$INSTALL_DIR" && printf "[4/4] ${DIM}Installed the ${CLI_NAME}${RESET}\n" >&2
```

File: install.sh [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+  fi
+}
+
+install_release() {
+  printf "[1/4] ${DIM}Downloading Hyp CLI from NPM${RESET}\n"
```
Comment (mattjohnsonpint) [inline install.sh:183]:
```suggestion
  printf "[1/4] ${DIM}Downloading the ${CLI_NAME} from NPM${RESET}\n"
```

File: install.sh [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+
+  local url="https://registry.npmjs.org/${PACKAGE_ORG:+$PACKAGE_ORG/}${PACKAGE_NAME}/-/${PACKAGE_NAME}-${VERSION}.tgz"
+
+  download_archive="$(
+    download_release "$url"
+    exit "$?"
+  )"
+  exit_status="$?"
+  if [ "$exit_status" != 0 ]; then
+    printf "Could not download Hyp version '$VERSION' from\n$url\n" >&2
```
Comment (mattjohnsonpint) [inline install.sh:193]:
```suggestion
    printf "Could not download the ${CLI_NAME} version '$VERSION' from\n$url\n" >&2
```

File: install.sh [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+    exit 1
+  fi
+
+  printf "$CLEAR_LINE" >&2
+  printf "[1/4] ${DIM}Downloaded latest Hyp CLI v${VERSION}${RESET}\n" >&2
```
Comment (mattjohnsonpint) [inline install.sh:198]:
```suggestion
  printf "[1/4] ${DIM}Downloaded latest ${CLI_NAME} v${VERSION}${RESET}\n" >&2
```

File: install.sh [hunk 11]
```diff
@@ -0,0 +1,290 @@
...
+
+  # make sure npm can be found too
+  if [ -z "$NPM_PATH" ]; then
+    printf "${RED}npm is not installed. Please install npm and try again.${RESET}\n" >&2
+    exit 1
+  fi
+}
+
+# This is the entry point
+printf "\n${BOLD}${BLUE}Hyp${RESET} Installer ${DIM}v${INSTALLER_VERSION}${RESET}\n\n" >&2
```
Comment (mattjohnsonpint) [inline install.sh:274]:
```suggestion
printf "\n${BOLD}${BLUE}${CLI_NAME}${RESET} Installer ${DIM}v${INSTALLER_VERSION}${RESET}\n\n" >&2
```

File: install.sh [hunk 12]
```diff
@@ -0,0 +1,290 @@
...
+
+check_platform
+check_node
+install_version
+
+printf "\nThe Hyp CLI has been installed! 🎉\n" >&2
```
Comment (mattjohnsonpint) [inline install.sh:280]:
```suggestion
printf "\nThe ${CLI_NAME} has been installed! 🎉\n" >&2
```

File: install.sh (hunk 4 above)
Comment (mattjohnsonpint) [reply install.sh:12]: @ryanfoxtyler should decide it it's "Hypermode CLI" or "Hyp CLI" before we merge this, but I like having it in a single variable.

**Review by mattjohnsonpint — CHANGES_REQUESTED** [review]

File: install.sh (hunk 4 above)
Comment (mattjohnsonpint) [reply install.sh:12]: I'll update the Modus installer with that too.

Comment (ryanfoxtyler) [reply install.sh:12]: Hyp CLI

Comment (mattjohnsonpint) [reply install.sh:12]: Ryan said "Hyp CLI". 👍

**Review by mattjohnsonpint — APPROVED** [review]

**Review by mattjohnsonpint — APPROVED** [review]

**Pull request merged**